subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -93,33 +93,79 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx, uint8_t phy)
 	struct pdu_adv *pdu;
 	uint8_t aux_handle;
 	uint8_t *ptr;
+	uint8_t phy;
 
 	ftr = &rx->rx_ftr;
-	lll = ftr->param;
-	aux = (void *)HDR_LLL2EVT(lll);
-	if (((uint8_t *)aux < (uint8_t *)ll_scan_aux_pool) ||
-	    ((uint8_t *)aux > ((uint8_t *)ll_scan_aux_pool +
-			    (sizeof(struct ll_scan_aux_set) *
-			     (CONFIG_BT_CTLR_SCAN_AUX_SET - 1))))) {
-		/* NOTE: We are most probably ADV_EXT_IND on primary channel */
-		aux = NULL;
-	}
 
-	scan = NULL;
-	sync = NULL;
-#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC)
-	if (aux) {
-		struct lll_scan *lll_scan;
-
-		lll_scan = aux->rx_head->rx_ftr.param;
-		scan = (void *)HDR_LLL2EVT(lll_scan);
-	} else {
+	switch (rx->type) {
+	case NODE_RX_TYPE_EXT_1M_REPORT:
+		lll = NULL;
+		aux = NULL;
 		scan = (void *)HDR_LLL2EVT(ftr->param);
+		sync = scan->per_scan.sync;
+		phy = BIT(0);
```

Comment:
Sorry about all the macro suggestions, but I usually find macros easier to understand and read, than these literals: We could also add `define`s for the different PHYs